src/main/java/it/rate/webapp/services/GoogleImageService.java

```diff
@@ -25,19 +29,28 @@ public class GoogleImageService implements ImageService {
   private String UPLOAD_DIRECTORY;
 
   private final Drive driveService;
-
+  private final ImageEditor imageEditor;
 
   @Override
-  public String saveImage(MultipartFile image) throws IOException {
+  public String saveImage(@NotNull MultipartFile image, String userName)
+      throws IOException { // todo: get principal
 
     FileContent mediaContent;
-    String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();
 
-    Path tempFilePath = Files.createTempFile("temp_" + currentUser, null);
+    String originalFileName = image.getOriginalFilename();
+
+    String fileExtension = getFileExtension(originalFileName);
+
+    if (fileExtension.isEmpty()) {
+      fileExtension = ".jpg";
+    }
+    Path tempFilePath = Files.createTempFile("temp_" + userName, "." + fileExtension);

```

Comment:
Everything looks good, I would just double-check if the Files.createTempFile needs to be deleted or if it will be deleted sometime in the future automatically

Reply:
If it stays there forever, it would be good to delete it after its not used anymore

Reply:
You were correct, files need to be deleted. Caught in try-catch block with finally